pkg/yaml/append_processor.go

```diff
@@ -0,0 +1,95 @@
+package yaml
+
+import (
+	"strings"
+)
+
+type AppendProcessor struct{}
+
+func NewAppendProcessor() *AppendProcessor {
+	return &AppendProcessor{}
+}
+
+func (ap *AppendProcessor) MergeWithAppend(dest, src map[string]any) error {
+	convertToStringMapInPlace(dest)
+	convertToStringMapInPlace(src)
+
+	for key, srcValue := range src {
+		if IsAppendKey(key) {
+			baseKey := GetBaseKey(key)
+			destValue, exists := dest[baseKey]
+			if exists {
+				if isSlice(srcValue) && isSlice(destValue) {
+					destSlice := destValue.([]any)
+					srcSlice := srcValue.([]any)
+					dest[baseKey] = append(destSlice, srcSlice...)
+				} else {
+					dest[baseKey] = srcValue
```

Comment:
Am I seeing this correctly? Is this just a hard overwrite of the value instead of a merge? 🫨